### PR TITLE
Update concurrency benchmarks to the latest streamly

### PR DESCRIPTION
**Pull request by adithyaov**
- Update stack to use lts-16.11 (ghc-8.8.3)
- Add a shell.nix
  - Pin nixpkgs at 5272327b81ed355bbed5659b8d303cf2979b6953
  - The default ghc being 8.6.5

### CODE REVIEW

File: concurrency-benchmarks.cabal [hunk 1]
```diff
@@ -54,25 +71,27 @@ benchmark benchmarks
     transformers        >= 0.4   && < 0.6,
     async               >= 2.1.1 && < 2.3,
     random              >= 1.0   && < 2,
-    streamly            >= 0.4.0 && < 0.6
+    streamly
```
Comment (harendra-kumar) [inline concurrency-benchmarks.cabal:74]: Are you sure all streamly versions work with this package?

File: Benchmarks.hs [hunk 2]
```diff
@@ -12,8 +12,9 @@ import System.Random (randomRIO)
 import Data.IORef
 
 import Gauge
-import Streamly
-import qualified Streamly.Prelude as S
+import Streamly.Prelude as S
```
Comment (harendra-kumar) [inline Benchmarks.hs:15]: `import qualified Streamly.Prelude as Stream`

File: Benchmarks.hs [hunk 3]
```diff
@@ -12,8 +12,9 @@ import System.Random (randomRIO)
...
+
+import Prelude as P
```
Comment (harendra-kumar) [inline Benchmarks.hs:17]: We can remove this once we import Streamly.Prelude qualified?

File: .github/workflows/packcheck.yml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        # The reason we have name here is to force an additional config instead
+        # of adding to an existing config while adding additional configs.
+        # Look at <https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix>
+        # for more info about adding matrix elements.
+        # Adding any element to the list will increase the number of matrix
+        # elemets proportional to the cross product.
+        name: [ci]
+        # Uncomment GHC versions you want the CI to run on.
+        ghc_version:
+          [ 8.8.4
```
Comment (harendra-kumar) [inline .github/workflows/packcheck.yml:102]: Can we add 8.10 as well? We do not need to build the charts with 8.10, but just build the code. Also, tested-with says 8.6.5, so we can add that too.

File: concurrency-benchmarks.cabal (hunk 1 above)
Comment (adithyaov) [reply concurrency-benchmarks.cabal:74]: No, I'm fixing the version of streamly via the env. `cabal.project`, `stack.yaml`. We can keep it rather simple. Developers will be the only ones using this. We don't need to support it over multiple versions.

**Comment by adithyaov** [comment]
Closed in favor of #2 
Much cleaner code and commits

**Pull request closed without merging**